IdentityCore/tests/integration/MSIDDefaultTokenCacheIntegrationTests.m

```diff
@@ -0,0 +1,604 @@
+}
+
+
+- (void)testSaveAccessToken_sameEverythingWithScopesDontIntersect_shouldWriteNewToken
+{
+    MSIDAccount *account = [[MSIDAccount alloc] initWithUpn:DEFAULT_TEST_ID_TOKEN_USERNAME
+                                                       utid:DEFAULT_TEST_UTID
+                                                        uid:DEFAULT_TEST_UID];
+    
+    MSIDToken *token1 = [[MSIDToken alloc] initWithTokenResponse:[MSIDTestTokenResponse v2DefaultTokenResponse]
+                                                         request:[MSIDTestRequestParams v2DefaultParams]
+                                                       tokenType:MSIDTokenTypeAccessToken];
+    
+    // save 1st token with default test scope
+    [_cacheAccessor saveAccessToken:token1
+                            account:account
+                      requestParams:[MSIDTestRequestParams v2DefaultParams]
+                            context:nil
+                              error:nil];
+    XCTAssertEqual([[_dataSource allDefaultAccessTokens] count], 1);
+    
+    // save 2nd token with non-intersecting scope
+    NSOrderedSet<NSString *> *scopes = [NSOrderedSet orderedSetWithObjects:@"profile.read", nil];
+    
+    MSIDToken *token2 = [[MSIDToken alloc] initWithTokenResponse:[MSIDTestTokenResponse v2DefaultTokenResponseWithScopes:scopes]
+                                                         request:[MSIDTestRequestParams v2DefaultParams]
+                                                       tokenType:MSIDTokenTypeAccessToken];
+    
+    NSError *error = nil;
+    
+    BOOL result = [_cacheAccessor saveAccessToken:token2
+                                          account:account
+                                    requestParams:[MSIDTestRequestParams v2DefaultParamsWithScopes:scopes]
+                                          context:nil
+                                            error:&error];
+    
+    XCTAssertNil(error);
+    XCTAssertTrue(result);
+    
+    NSArray *accessTokensInCache = [_dataSource allDefaultAccessTokens];
+    XCTAssertEqual([accessTokensInCache count], 2);
+    XCTAssertEqualObjects(accessTokensInCache[0], token1);
+    XCTAssertEqualObjects(accessTokensInCache[1], token2);
+}
+
+
+
+- (void)testSaveSharedRTForAccount_withRT_shouldSaveOneEntry
+{
+    MSIDAccount *account = [[MSIDAccount alloc] initWithUpn:DEFAULT_TEST_ID_TOKEN_USERNAME
+                                                       utid:DEFAULT_TEST_UTID
+                                                        uid:DEFAULT_TEST_UID];
+    
+    MSIDToken *token = [[MSIDToken alloc] initWithTokenResponse:[MSIDTestTokenResponse v2DefaultTokenResponse]
+                                                        request:[MSIDTestRequestParams v2DefaultParams]
+                                                      tokenType:MSIDTokenTypeRefreshToken];
+    
+    NSError *error = nil;
+    
+    BOOL result = [_cacheAccessor saveAccessToken:token
```

Comment:
test is called testSaveSharedRTForAccount, but it calls saveAccessToken

Reply:
😂